### DISCUSSION TITLE
Support of dynamic rendering of markdown and images

### DISCUSSION THREAD

**Discussion by Nauaho** [neovide/neovide#2663] [Ideas]
### Introduction
Neovim is awesome, yet it has its limitations. Yet recently I tried to work with Jupyter notebooks and, honestly, I was disappointed. I know about existence of Molten.nvim and Magma.nvim pluginsm but the limit Users to the specific terminal emulator and/or demand additional applications for assistance in rendering on User's system. Same story about documentation of Numpy: the documentation is showing up but it is written in Markdown, hence it's unreadable in some places. 

### Suggestion
Since Neovide is GUI application, I propose to utilize its rendering capabilities to a greater extent for dynamically rendering images and markdown. Of course if the implementation won't find a way "to fool" neovim that it has environment which implements kitty image protocol the Jupyter notebooks problem will stay, but who said that some plugins for neovim are the only option?

### Conclusion
In the end i propose to start from the "humble" beginings of elimination of using another window for preview of markdown and something that will also make Neovide somewhat light-weight alternative for Obsidian and other closed-source notetaking applications.

**Comment by kjy5** [comment]
I agree with this. As a GUI application it would be possible to handle situations that emacs can do (since it's a GUI application).

**Comment by m-dwork** [comment]
I was toying with this the other day and came across a few plugins for neovim that use ascii or other tricks.. but I wasn't able to get any to work in neovide.

https://github.com/adelarsq/image_preview.nvim
https://github.com/nvim-telescope/telescope-media-files.nvim

**Reply by Nauaho** [reply]
I actually tried image.nvim plugin and it warked... Partially because I use WSL and kitty terminal has no support for windows and works, at least from my experience justy bad, again, bevause of platform I use. 
Telescope-media-files: I didn't try, but in general I needed image support because of Jypiter Notebooks. 
And still some of those plugins are pretty demanding: Terminal Emulators, Operational Systems etc.

**Comment by justyn** [comment]
There's been some discussion of image support [here](https://github.com/neovide/neovide/issues/2088), although I don't know the latest status.

I agree that Neovide has the potential to render things like Markdown in a far superior way to that possible with Neovim in a terminal. As well as images, for example, it is pretty standard for markdown headers to be different font sizes indicating their level.

Not sure if this is something that is fundamentally against the principals of Neovide, or if it might be possible one day if someone implements it?
